src/apps/peer/udpserver.c

```diff
@@ -32,86 +32,123 @@
 #include "apputils.h"
 #include "stun_buffer.h"
 
+struct listen_info {
+  ioa_addr addr;
+  int received;
+  int verbose;
+  struct listen_info *next;
+};
+
 /////////////// io handlers ///////////////////
 
 static void udp_server_input_handler(evutil_socket_t fd, short what, void *arg) {
 
   if (!(what & EV_READ))
     return;
 
-  ioa_addr *addr = (ioa_addr *)arg;
+  struct listen_info *listen = (struct listen_info *)arg;
 
   int len = 0;
-  int slen = get_ioa_addr_len(addr);
-  stun_buffer buffer;
-  ioa_addr remote_addr;
+  int slen = get_ioa_addr_len(&listen->addr);
+  stun_buffer *buffer = (stun_buffer *)malloc(sizeof(stun_buffer));
+  ioa_addr remote_addr = {0};
+  if (!buffer)
+    return;
 
   do {
-    len = recvfrom(fd, buffer.buf, sizeof(buffer.buf) - 1, 0, (struct sockaddr *)&remote_addr, (socklen_t *)&slen);
+    len = recvfrom(fd, buffer->buf, sizeof(buffer->buf) - 1, 0, (struct sockaddr *)&remote_addr, (socklen_t *)&slen);
   } while (len < 0 && socket_eintr());
 
-  buffer.len = len;
+  buffer->len = len;
 
   if (len >= 0) {
+    listen->received += len;
+    if (listen->verbose) {
+      uint8_t ra[64] = {0}, la[64] = {0};
```

Comment:
magic numbers. Please use a named constant or some kind of sizeof() calculation.

Also please don't declare multiple variables on the same line, it makes it impossible to understand what's going on